### PR TITLE
Only activate request context in RESTEasy Reactive if it's not already active

### LINKED ISSUE #34693: Vertx RouteFilter not propagating request context to a resteasy-reactive resource
### Describe the bug

When having  a `@RouteFilter` setting information on a `@RequestScoped` bean, the bean is not being propagated correctly and the information is lost as a new request scope is beeing openend.

### Expected behavior

The request scope of an  `@RouteFilter` should be propagated to my `jakarta.ws` rest controller.

### Actual behavior

A new request scope is beeing opened between my `@RouteFilter` and the invocation of my rest controller.

```
2023-07-12 10:24:06,341 INFO  [io.quarkus] (main) code-with-quarkus 1.0.0-SNAPSHOT on JVM (powered by Quarkus 3.1.2.Final) started in 2.462s. Listening on: http://localhost:8081
2023-07-12 10:24:06,367 INFO  [io.quarkus] (main) Profile test activated. 
2023-07-12 10:24:06,367 INFO  [io.quarkus] (main) Installed features: [cdi, jacoco, reactive-routes, resteasy-reactive, smallrye-context-propagation, vertx]
2023-07-12 10:24:07,578 WARN  [org.acm.iss.RequestScopedBean] (vert.x-eventloop-thread-0) ## New RequestScopedBean created
2023-07-12 10:24:07,579 WARN  [org.acm.iss.VertxRoutFilter] (vert.x-eventloop-thread-0) ## value from field 'info' of request scoped bean: testParamValue 
2023-07-12 10:24:07,584 WARN  [org.acm.iss.ExampleRestController] (executor-thread-1) ## Handeling getRequetScopedInfo
2023-07-12 10:24:07,584 WARN  [org.acm.iss.RequestScopedBean] (executor-thread-1) ## New RequestScopedBean created
2023-07-12 10:24:07,584 WARN  [org.acm.iss.ExampleRestController] (executor-thread-1) ## value from field 'info' of request scoped bean: null
2023-07-12 10:24:07,587 WARN  [org.acm.iss.RequestScopedBean] (executor-thread-1) ## RequestScopedBean destroyed
2023-07-12 10:24:07,587 WARN  [org.acm.iss.RequestScopedBean] (executor-thread-1) ## RequestScopedBean destroyed
2023-07-12 10:24:07,694 INFO  [io.quarkus] (main) code-with-quarkus stopped in 0.027s
```

### How to Reproduce?

https://github.com/HerrDerb/quarkus-issue/tree/vertx-requestscope-issue

### PR DESCRIPTION
**Pull request by geoand**
Fixes: #34693

TODO: Add a test covering the case mentioned in the original issue

### CODE REVIEW

File: extensions/resteasy-reactive/quarkus-resteasy-reactive-common/runtime/src/main/java/io/quarkus/resteasy/reactive/common/runtime/ArcThreadSetupAction.java [hunk 1]
```diff
@@ -15,7 +15,9 @@ public ArcThreadSetupAction(ManagedContext managedContext) {
 
     @Override
     public ThreadState activateInitial() {
-        managedContext.activate();
+        if (!managedContext.isActive()) {
```
Comment (mkouba) [inline extensions/resteasy-reactive/quarkus-resteasy-reactive-common/runtime/src/main/java/io/quarkus/resteasy/reactive/common/runtime/ArcThreadSetupAction.java:18]: I think that you also need to make the returned `ThreadState` a no-op.

Comment (geoand) [reply extensions/resteasy-reactive/quarkus-resteasy-reactive-common/runtime/src/main/java/io/quarkus/resteasy/reactive/common/runtime/ArcThreadSetupAction.java:18]: I don't think so, but I'll have to look closer

Comment (mkouba) [reply extensions/resteasy-reactive/quarkus-resteasy-reactive-common/runtime/src/main/java/io/quarkus/resteasy/reactive/common/runtime/ArcThreadSetupAction.java:18]: Well, if you don't activate the context then you should not destroy it either :shrug:

Comment (geoand) [reply extensions/resteasy-reactive/quarkus-resteasy-reactive-common/runtime/src/main/java/io/quarkus/resteasy/reactive/common/runtime/ArcThreadSetupAction.java]: Let's see what this latest change does.

I still need to add an test that uses Reactive Routes and RESTEasy Reactive

**Comment by geoand** [comment]
This actually needs more thought

**Comment by geoand** [comment]
I am going to close this as it was actually implemented in https://github.com/quarkusio/quarkus/pull/40408

**Pull request closed without merging**